src/hpp/corbaserver/rbprm/__main__.py

```diff
@@ -31,27 +33,32 @@
         print("Cannot import " + scenario_name + ". Check if the path is correct")
         sys.exit(1)
 
-# kill already existing instance of the viewer/server
-subprocess.run(["killall", "gepetto-gui"])
+# kill already existing instance of the server
 subprocess.run(["killall", "hpp-rbprm-server"])
-# run the viewer/server in background
+# run the server in background :
 # stdout and stderr outputs of the child process are redirected to devnull (hidden).
 # preexec_fn is used to ignore ctrl-c signal send to the main script (otherwise they are forwarded to the child process)
-process_viewer = subprocess.Popen("gepetto-gui",
-                                  stdout=subprocess.PIPE,
-                                  stderr=subprocess.DEVNULL,
-                                  preexec_fn=os.setpgrp)
 process_server = subprocess.Popen("hpp-rbprm-server",
                                   stdout=subprocess.PIPE,
                                   stderr=subprocess.DEVNULL,
                                   preexec_fn=os.setpgrp)
+# register cleanup methods to kill server when exiting python interpreter
+atexit.register(process_server.kill)
+
+# do the same for the viewer, exept if --no-viewer flag is set
+disable_viewer = args.no_viewer
+if disable_viewer is None:
```

Comment:
`action="store_true"` shoud set `args.no_viewer` to either True or False, so I suggest:
`if not args.no_viewer:`